pyteal/compiler/compiler.py

```diff
@@ -128,9 +128,16 @@ def compileSubroutine(
 
     if not ast.has_return():
         if ast.type_of() == TealType.none:
-            ast = Seq([ast, Return()])
+            ret_expr = Return()
+            ret_expr.trace = ast.trace
+            seq_expr = Seq([ast, ret_expr])
+            seq_expr.trace = ret_expr.trace
+            ast = seq_expr
         else:
-            ast = Return(ast)
+            ret_expr = Return()
+            ret_expr.trace = ast.trace
+            ret_expr.value = ast
```

Comment:
Is this explicit setting of value a work around `Return` constructor's `require_type` check?